src/java/org/apache/cassandra/streaming/StreamManager.java

```diff
@@ -213,6 +226,98 @@ private static double calculateEffectiveRateInBytes(double throughput)
     private final Map<UUID, StreamResultFuture> initiatorStreams = new NonBlockingHashMap<>();
     private final Map<UUID, StreamResultFuture> followerStreams = new NonBlockingHashMap<>();
 
+    private final Map<UUID, StreamingState> states = new NonBlockingHashMap<>();
+    private volatile ScheduledFuture<?> cleanup = null;
+
+    public StreamManager()
+    {
+        addListener(new StreamListener()
+        {
+            @Override
+            public void onRegister(StreamResultFuture result)
+            {
+                StreamingState state = new StreamingState(result);
+                StreamingState previous = states.putIfAbsent(state.getId(), state);
```

Comment:
still wondering if possible to have 1 node be a follower and not; doesn't make sense to stream to yourself, but not sure if that is actually blocked or not... wondering if I should make this `(id, follower)`